### PR TITLE
New branch with Week 4 commits

### PR DESCRIPTION
**Pull request by evd123**
I reverted Pull Request #2 (WebApp Updates) to before making the Maps API changes and moved those changes to this branch and pull request.

### CODE REVIEW

**Review by lcatherine — CHANGES_REQUESTED** [review]
There are some spacing / newline issues throughout. 

Also you can remove the ToDo for handling comments since it has been handled :D

**Comment by lcatherine** [comment]
It looks like there are conflicts, please resolve before merging

File: portfolio/src/main/java/com/google/sps/servlets/TravelDataServlet.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    for (Entity entity: results.asIterable()) {
+        String continent = (String) entity.getProperty("continent");
+        strContinents.add(continent);
+    }
+
+    HashMap<String, Integer> voteCount = getVotes(strContinents);
+    
+    response.setContentType("application/json");
+    Gson gson = new Gson();
+    response.getWriter().println(gson.toJson(voteCount));
```
Comment (lyze) [inline portfolio/src/main/java/com/google/sps/servlets/TravelDataServlet.java:55]: gson.ToJson(voteCount, gson.newJsonWriter(response.getWriter());

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 2]
```diff
@@ -11,22 +11,94 @@
...
+      String email = (String) entity.getProperty("email");
+      long timestamp = (long) entity.getProperty("timestamp");
+
+      Message curMessage = new Message(id, comment, yes, email, timestamp);
+      allMessages.add(curMessage);
+    }
+    // Send the JSON as the response
+    response.setContentType("application/json");
+    Gson gson = new Gson();
+    response.getWriter().println(gson.toJson(allMessages));
```
Comment (lyze) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:58]: gson.toJson(allMessages, gson.newJsonWriter(response.getWriter()));

**Review by lyze — CHANGES_REQUESTED** [review]

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 3]
```diff
@@ -11,22 +11,94 @@
...
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    response.setContentType("text/html;");
-    response.getWriter().println("<h1>Hello world!</h1>");
+    Query query = new Query("Message").addSort("timestamp", SortDirection.DESCENDING);
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    PreparedQuery results = datastore.prepare(query);
+
+    List < Message > allMessages = new ArrayList < > ();
```
Comment (lcatherine) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:44]: weird spacing around the angled brackets

File: portfolio/src/main/java/com/google/sps/servlets/DataServlet.java [hunk 4]
```diff
@@ -11,22 +11,94 @@
...
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(messageEntity);
+
+    // Redirect back to the HTML page.
+    response.sendRedirect("/index.html");
+  }
+
+  /*
+  Function to check if email is submitted if the user wants a response. Don't think this function actually does anything.
```
Comment (lcatherine) [inline portfolio/src/main/java/com/google/sps/servlets/DataServlet.java:93]: "Don't think this function actually does anything." 🤔  So should this be removed or fixed?

File: portfolio/src/main/java/com/google/sps/servlets/TravelDataServlet.java [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+import com.google.appengine.api.datastore.Entity;
+import com.google.appengine.api.datastore.PreparedQuery;
+import com.google.appengine.api.datastore.Query;
+
+import com.google.gson.Gson;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.HashMap;
+import java.util.Map;
```
Comment (lcatherine) [inline portfolio/src/main/java/com/google/sps/servlets/TravelDataServlet.java:28]: unused

File: portfolio/src/main/java/com/google/sps/servlets/TravelDataServlet.java [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+import java.util.Scanner;
```
Comment (lcatherine) [inline portfolio/src/main/java/com/google/sps/servlets/TravelDataServlet.java:29]: same

**Pull request merged**